core/network/src/main/kotlin/com/google/samples/apps/nowinandroid/core/network/di/NetworkModule.kt

```diff
@@ -72,20 +75,21 @@ internal object NetworkModule {
     @Provides
     @Singleton
     fun imageLoader(
-        okHttpCallFactory: Call.Factory,
+        // We specifically request dagger.Lazy here, so that it's not instantiated from Dagger.
+        okHttpCallFactory: dagger.Lazy<Call.Factory>,
         @ApplicationContext application: Context,
-    ): ImageLoader = ImageLoader.Builder(application)
-        .callFactory(okHttpCallFactory)
```

Comment:
It looks like this is the culprit of the slow initialization, as the `OkHttpClient` is created eagerly instead of using `ImageLoader.Builder.callFactory(() -> Call.Factory)`, which accepts a callback that's invoked on a background thread. It looks like this PR also handles that fix using a `dagger.Lazy`.

I'd recommend migrating to `ImageLoader.Builder.callFactory { }` instead of creating the `ImageLoader` asynchronously, as that should fix the performance issue. I included a quick benchmark [here](https://github.com/coil-kt/coil/issues/2097#issuecomment-1925892114). Also added more context [here](https://github.com/coil-kt/coil/issues/2097#issuecomment-1922061816)!